modules/network/src/main/java/org/apache/ignite/network/DefaultMessagingService.java

```diff
@@ -110,18 +115,20 @@ public DefaultMessagingService(
             TopologyService topologyService,
             StaleIdDetector staleIdDetector,
             ClassDescriptorRegistry classDescriptorRegistry,
-            UserObjectMarshaller marshaller
+            UserObjectMarshaller marshaller,
+            CriticalWorkerRegistry criticalWorkerRegistry
     ) {
         this.factory = factory;
         this.topologyService = topologyService;
         this.staleIdDetector = staleIdDetector;
         this.classDescriptorRegistry = classDescriptorRegistry;
         this.marshaller = marshaller;
+        this.criticalWorkerRegistry = criticalWorkerRegistry;
 
-        this.outboundExecutor = Executors.newSingleThreadExecutor(NamedThreadFactory.create(nodeName, "MessagingService-outbound-", LOG));
+        this.outboundExecutor = new CriticalSingleThreadExecutor(NamedThreadFactory.create(nodeName, "MessagingService-outbound", LOG));
```

Comment:
Why did you remove the "-" at the end?

Reply:
Because `NamedThreadFactory#threadPrefix()` adds the dash itself

Reply:
Oh, looks like I missed that